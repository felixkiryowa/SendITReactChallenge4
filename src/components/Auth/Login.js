import React from 'react';
import './auth.scss';

const  Login = (
    {
        handleSubmit,
        handlePasswordChange,
        handleUsernameChange,
        username,
        password,
        errors,

    }
) =>  {
    return (
      
        <div className="card card-container">
            <img id="profile-img" className="profile-img-card" src="//ssl.gstatic.com/accounts/ui/avatar_2x.png" />
            <p id="profile-name" className="profile-name-card"></p>
            <form className="form-signin" onSubmit={ handleSubmit }>
                <span id="reauth-email" className="reauth-email"></span>
                <input 
                    type="text" 
                    id="inputEmail" 
                    className="form-control" 
                    placeholder="Enter Username"
                    name="username"
                    onChange={ handleUsernameChange }
                    value={ username }
                    required autoFocus 
                />
                <br/>
                <input 
                    type="password" 
                    id="inputEmail" 
                    className="form-control" 
                    placeholder="Enter Password"
                    value={ password }
                    onChange={ handlePasswordChange }
                    name="password"
                    required 
                 />                
                <button 
                    className="btn btn-lg btn-primary btn-block btn-signin login-btn" 
                    type="submit">
                    Sign in
                </button>
            </form>
            <a href="#" className="forgot-password">
                Forgot the password?
            </a>
        </div>
  
    )
}

export default Login;

