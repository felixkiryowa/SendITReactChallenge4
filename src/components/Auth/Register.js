import React from 'react';
import './auth.scss';

const Register = (
    {
        handleChange,
        handleSubmit,
        errors,
    }
) => {
	return (
		<section className="testimonial py-5" id="testimonial">
            <div className="container reg-container">
				<div className="row ">
					<div className="col-md-4 py-5 custom-card-color text-white text-center ">
						<div className=" ">
							<div className="card-body">
								<h2 className="py-3">Registration</h2>
								<p>Create An Account With Send-IT such that your able to create parcel orders</p>
							</div>
						</div>
					</div>
					<div className="col-md-8 py-5 border">
						<h4 className="pb-4">Please fill with your details</h4>
						<form onSubmit={handleSubmit}>
							<div className="form-row">
								<div className="form-group col-md-6">
									<input
										name="firstname"
                                        id="FirstName"
										placeholder="First Name"
										className="form-control"
										type="text"
                                        onChange={handleChange}
									/>
								</div>
								<div className="form-group col-md-6">
									<input 
                                    type="text" 
                                    className="form-control" 
                                    name="username" 
                                    placeholder="Username" 
                                    onChange={handleChange}
                                    />
								</div>
							</div>
							<div className="form-row">
								<div className="form-group col-md-6">
									<input
										name="lastname"
										placeholder="Last Name"
										className="form-control"
										required="required"
                                        onChange={handleChange}
										type="text"
									/>
								</div>
								<div className="form-group col-md-6">
									<input
										name="password"
										placeholder="Password"
										className="form-control"
										required="required"
                                        onChange={handleChange}
										type="password"
									/>
								</div>
							</div>
							<div className="form-row">
								<div className="form-group col-md-6">
									<input
										name="contact"
										placeholder="Mobile No."
										className="form-control"
										required="required"
                                        onChange={handleChange}
										type="text"
                                        maxLength="10"
									/>
                                    {
                                        errors.contact && (<
                                            div style={{ color: 'red' }}>{errors.contact}</div>)
                                    }
								</div>
								<div className="form-group col-md-6">
									<input
										name="confirmPassword"
										placeholder="Confirm Password"
										className="form-control"
										required="required"
                                        onChange={handleChange}
										type="password"
									/>
                                    {
                                        errors.confirmPassword && (<
                                        div style={{ color: 'red' }}>{errors.confirmPassword}</div>)
                                    }
								</div>
							</div>
							<div className="form-row">
								<div className="form-group col-md-6">
									<input
										name="useremail"
										placeholder="Email e.g daddyfelix56@gmail.com"
										className="form-control"
										required="required"
                                        onChange={handleChange}
										type="email"
									/>
								</div>
							</div>
                            <br/>
                            <br/>
							<div className="form-row">
                                <div className="form-group col-md-12">
                                    <button type="submit" className="btn  float-right reg-button">
                                        Sign Up
								    </button>
                                </div>
							</div>
						</form>
					</div>
				</div>
			</div>
		</section>
	);
};

export default Register;
