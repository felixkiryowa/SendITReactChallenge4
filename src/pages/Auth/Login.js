import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { signInAction } from '../../actions/authActions';
import { connect } from 'react-redux';
import LoginPage from '../../components/Auth/Login';


export class Login extends Component {

    constructor() {
        super();
        this.state = {
            username:'',
            password: '',
            errors : {

            }
        }

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleUsernameChange = this.handleUsernameChange.bind(this);
        this.handlePasswordChange = this.handlePasswordChange.bind(this);
    }

    handleSubmit = (event) => {
        event.preventDefault();
        const { loginUser } = this.props;
        const loginData = {
            "username": this.state.username,
            "password": this.state.password
        }
        loginUser(loginData, this.props);
    }

    handleUsernameChange = (event) => {
        event.preventDefault();
        this.setState({ username: event.target.value });
    }

    handlePasswordChange = (event) => {
        event.preventDefault();
        this.setState({ password: event.target.value });
    }

    render() {
        return (
            <div>
                <LoginPage
                    handleSubmit= {this.handleSubmit}
                    handlePasswordChange ={ this.handlePasswordChange }
                    handleUsernameChange= { this.handleUsernameChange }
                    errors={ this.state.errors }
                    username={ this.state.username }
                    password={ this.state.password }
                 />
            </div>
        )
    }
}

Login.propTypes = {
    loginUser: PropTypes.func.isRequired,
    history: PropTypes.shape({
        push: PropTypes.func.isRequired,
    }).isRequired,
};

export default connect(null, {
    loginUser: signInAction
})(Login);
