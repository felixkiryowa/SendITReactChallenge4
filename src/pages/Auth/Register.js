import React, { Component } from 'react'
import { connect } from 'react-redux';
import { signupAction } from '../../actions/authActions';
import PropTypes from 'prop-types';
import RegisterComponent from '../../components/Auth/Register';


class Register extends Component {

    constructor() {
        super();
        this.state = {
            useremail: '',
            username: '',
            password: '',
            confirmPassword: '',
            firstname: '',
            lastname: '',
            contact: '',
            errors: {

            }
        }
    }

    handleChange = (event) => {
        event.preventDefault();
        this.setState({
            [event.target.name]: event.target.value,
        });
    }

    checkPasswordMatch = (password, confirmPassword) => {
        if (password === confirmPassword) {
            this.setState({
                errors: {}
            })
        } else {
            this.setState({
                errors: {
                    confirmPassword: "Passwords do not match"
                }
            })
        }
    }

    checkValidPhoneNumber = (phoneNumber) => {
        var alphaExp = /^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/;
        if (phoneNumber.match(alphaExp)) {
            this.setState({
                errors: {}
            })
        } else {
            this.setState({
                errors: {
                    contact: "Wrong Format, It should be 0700162509"
                }
            })
        }
    }

    handleSubmit = (event) => {
        event.preventDefault();
        const { signupUser } = this.props;
        const newUser = {
            "first_name": this.state.firstname,
            "last_name": this.state.lastname,
            "email": this.state.useremail,
            "phone_contact": this.state.contact,
            "username": this.state.username,
            "user_password": this.state.password,
            "user_type": "user"
        }
        this.checkPasswordMatch(this.state.password, this.state.confirmPassword);
        this.checkValidPhoneNumber(this.state.contact);
        signupUser(newUser, this.props);
        console.log(newUser);
    }

    render() {
        return (
            <div>
                <RegisterComponent
                 handleChange={this.handleChange}
                 handleSubmit={this.handleSubmit}
                 errors={this.state.errors}
                />
            </div>
        )
    }
}

Register.propTypes = {
    signupUser: PropTypes.func.isRequired,
    history: PropTypes.shape({
        push: PropTypes.func.isRequired,
    }).isRequired,
};


export default connect(null, {
    signupUser: signupAction
})(Register)

