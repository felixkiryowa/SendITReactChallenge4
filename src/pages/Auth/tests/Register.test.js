import React from 'react';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import { Register } from '../../Auth/Register';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const initialState = {
    authReducer: {},
};
const store = mockStore(initialState);


describe('Register Page Component', () => {
    let mountApp;
    it('Should render without crashing', () => {
        const props = {
            signupUser: jest.fn(),
            handleSubmit: jest.fn(),
            handleChange: jest.fn(),
            checkPasswordMatch: jest.fn(),
            checkValidPhoneNumber: jest.fn(),

        };
        mountApp = mount(
            <Router>
                <Provider store={store}>
                    <Register {...props} />
                </Provider>
            </Router>,
        );
        expect(mountApp).toMatchSnapshot();
    });

    it('should call an onChange function', () => {
        const props = {
            signupUser: jest.fn(),
            handleSubmit: jest.fn(),
            handleChange: jest.fn(),
            checkPasswordMatch: jest.fn(),
            checkValidPhoneNumber: jest.fn(),

        };
        const event = {
            preventDefault: () => { },
            target: {
                name: 'body',
                value: 'abc',
            },
        };
        const wrapper = mount(
            <Register {...props} />,
        );
        const firstnameInput = wrapper.find('#FirstName').first();
        firstnameInput.simulate('change', event);
        expect(event.target.value).toEqual('abc');
    });

    it('should call a submit a form', () => {
        const props = {
            signupUser: jest.fn(),
            handleSubmit: jest.fn(),
            handleChange: jest.fn(),
            checkPasswordMatch: jest.fn(),
            checkValidPhoneNumber: jest.fn(),

        }; 
        const wrapper = mount(
            <Register {...props} />,
        );
        const instance = wrapper.instance();
        const checkPasswordMatch = jest.spyOn(instance, 'checkPasswordMatch');
        const checkValidPhoneNumber = jest.spyOn(instance, 'checkValidPhoneNumber');
        instance.handleSubmit({ preventDefault: jest.fn() });
        expect(props.signupUser).toHaveBeenCalled();
        expect(checkPasswordMatch).toHaveBeenCalled();
        expect(checkValidPhoneNumber).toHaveBeenCalled();
    });

});
