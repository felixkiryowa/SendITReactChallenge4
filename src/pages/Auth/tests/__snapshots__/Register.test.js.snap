// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page Component Should render without crashing 1`] = `ReactWrapper {}`;
